File: mo-common/mo-common-mybatis/src/main/java/ink/ckx/mo/common/mybatis/result/PageResult.java
package ink.ckx.mo.common.mybatis.result;

import com.baomidou.mybatisplus.core.metadata.IPage;
import ink.ckx.mo.common.core.result.ResultCode;
import lombok.Data;

import java.io.Serializable;
import java.util.List;

/**
 * @author chenkaixin
 * @description
 * @since 2023/10/19
 */
@Data
public class PageResult<T> implements Serializable {

    private String code;

    private Data<T> data;

    private String msg;

    public static <T> PageResult<T> success(IPage<T> page) {
        PageResult<T> result = new PageResult<>();
        result.setCode(ResultCode.SUCCESS.getCode());

        Data<T> data = new Data<T>();
        data.setList(page.getRecords());
        data.setTotal(page.getTotal());

        result.setData(data);
        result.setMsg(ResultCode.SUCCESS.getMsg());
        return result;
    }

    @lombok.Data
    public static class Data<T> {

        private List<T> list;

        private long total;
    }
}

File: mo-common/mo-common-mybatis/src/main/kotlin/ink/ckx/mo/common/mybatis/result/PageResult.kt
package ink.ckx.mo.common.mybatis.result

import com.baomidou.mybatisplus.core.metadata.IPage
import ink.ckx.mo.common.core.result.ResultCode
import java.io.Serializable

/**
 * @author chenkaixin
 * @description
 * @since 2023/10/19
 */
data class PageResult<T>(
    var code: String,
    var data: Data<T>,
    var msg: String,
) : Serializable {
    companion object {
        @JvmStatic
        fun <T> success(page: IPage<T>): PageResult<T> {
            return PageResult(ResultCode.SUCCESS.code, Data(page.records, page.total.toInt()), ResultCode.SUCCESS.msg)
        }
    }
}

data class Data<T>(
    var list: List<T>,
    var total: Int
) : Serializable